src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,618 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.integration
+import data.real.ereal
+
+/-!
+
+# Vector valued measures
+
+This file defines vector valued measures, which are σ-additive functions from a set to a module `M`
+over the ring `R` such that it maps the empty set and non-measurable sets to zero. In the case
+that `R = M = ℝ`, we called the vector measure a signed measure and write `signed_measure α`.
+Similarly, when `R = M = ℂ`, we call the measure a complex measure and write `complex_measure α`.
+
+## Main definitions
+
+* `vector_measure` is a vector valued, σ-additive function that maps the empty
+  and non-measurable set to zero.
+
+## Implementation notes
+
+We require all non-measurable sets to be mapped to zero in order for the extensionality lemma
+to only compare the underlying functions for measurable sets.
+
+We use `has_sum` instead of `tsum` in the definition of vector measures in comparison to `measure`
+since this provides summablity. In the case of `signed_measure`, this is not necessary and one
+can construct a signed measure without proving summable using `signed_measure'.to_signed_measure`.
+
+## Tags
+
+vector measure, signed measure, complex measure
+-/
+
+noncomputable theory
+
+open_locale classical big_operators nnreal ennreal
+
+variables {α β : Type*} [measurable_space α]
+
+/-- A signed measure on a measurable space `α` is a σ-additive, real-valued function
+that maps the empty set to zero. -/
+structure vector_measure (α : Type*) [measurable_space α] (R : Type*) [ring R]
```

Comment:
I don't understand the role of `R` here. Can't you just remove it and speak of `vector_measure α E`? I have made the change and the file compiles perfectly well.

Reply:
You're right. I assumed a vector measure required `M` to be a module but it certainly does not